spec/lcm/integration/user_provisioning_end_to_end_spec.rb

```diff
@@ -0,0 +1,215 @@
+          auth_token: LcmConnectionHelper.environment[:prod_token],
+          environment: 'TESTING'
+        }
+        project_helper = Support::ProjectHelper.create(opts)
+        puts 'creating LDM and stuff'
+        project_helper.create_ldm
+        project_helper.load_data
+        puts "created project #{project_helper.project.pid}"
+        project_helper.project
+      end
+    end
+    @test_context = {
+      config: LcmConnectionHelper.environment,
+      s3_bucket: Support::S3Helper::BUCKET_NAME,
+      s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+      s3_key: 'user_data'
+    }
+    @ub_template_path = File.expand_path('../params/users_brick_e2e.json.erb', __FILE__)
+    @ufb_template_path = File.expand_path('../params/user_filters_brick_e2e.json.erb', __FILE__)
+
+    @domain = @rest_client.domain(LcmConnectionHelper.environment[:prod_organization])
+    @data_product = @domain.data_products('default')
+    @master_project = @rest_client.create_project(title: "Test MASTER project for #{@suffix}", auth_token: LcmConnectionHelper.environment[:prod_token])
+    @segment = @data_product.create_segment(segment_id: "testing-segment-#{@suffix}", master_project: @master_project)
+
+    @users = Array.new($user_amount) do |n|
+      "lcm-userprov-e2e-UPPERCASE-#{n}@gooddata.com"
+    end
+
+    @test_context = {
+      project_id: @projects.first.pid, # this doesn't really matter for the runtime
+      config: LcmConnectionHelper.environment,
+      s3_bucket: Support::S3Helper::BUCKET_NAME,
+      s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+      s3_key: 'user_filters',
+      sync_mode: 'sync_multiple_projects_based_on_pid',
+      data_product: 'default',
+      users_brick_input: {
+        s3_bucket: Support::S3Helper::BUCKET_NAME,
+        s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+        s3_key: 'users_brick_input'
+      }
+    }
+  end
+
+  describe '1 - initial MUF setup' do
+    before do
+      ensure_users_in_domain @users
+
+      $user_data = @projects.map do |c|
+        @users.map do |u|
+          {
+            custom_login: u,
+            project_id: c.pid,
+            position: 'developer',
+          }
+        end
+      end.flatten
+      users_csv = ConfigurationHelper.csv_from_hashes($user_data)
+      Support::S3Helper.upload_file(users_csv, @test_context[:users_brick_input][:s3_key])
+
+      atts = @projects.first.attributes.to_a
+      $attribute = atts.shift while !$attribute || $attribute.date_attribute? || !$attribute.primary_label
+      label = $attribute.primary_label
+      muf_template = {}
+      element = label.values.find { |v| !v[:value].empty? }
+      muf_template[label.title] = element[:value]
+      $mufs = $user_data.map do |u|
+        m = muf_template.dup
+        m[:login] = u[:custom_login]
+        m[:project_id] = u[:project_id]
+        m
+      end.flatten
+
+      upload_user_filters_csv($mufs)
+    end
+
+    it 'setups the MUFs' do
```

Comment:
Similar code found in user_provisioning_end_to_end_spec.rb:116, user_provisioning_end_to_end_spec.rb:190 (mass = 100)